src/stk/molecular/molecules/building_block.py

```diff
@@ -644,15 +653,53 @@ def _normalize_placer_ids(self, placer_ids, functional_groups):
         """
 
         if placer_ids is not None:
-            return placer_ids
+            return frozenset(placer_ids)
 
         if functional_groups:
-            return tuple(flatten(
+            return frozenset(flatten(
```

Comment:
```python
if functional_groups
```

won't work if `functional_groups` is an Iterable.

Reply:
Are you thinking `if len(functional_groups) > 0`:

Reply:
That won't work with an iterable.